src/accelerate/accelerator.py

```diff
@@ -1643,6 +1643,7 @@ def _prepare_ipex(self, *args):
                 "Trying to use IPEX but IPEX is not installed or IPEX's version does not match current PyTorch, please refer"
                 " to https://github.com/intel/intel-extension-for-pytorch."
             )
+            return args
```

Comment:
I personally feel this should raise a full `ImportError` error, rather than just a `warning` and not return anything. Not doing so led to some subtle bugs we didn't know existed during the transformers integration, and iirc transformers also raises an error.

Reply:
I agree during the initial PR I had some tests which were very slow and failing  (in 1118). 
I think we can keep as it is without returning and terminate in a  ImportError .

Reply:
currently. _prepare_ipex is only called when self.state.ipex_plugin is not None now. so maybe this import check should be removed

Reply:
No, it should stay and not warn but instead raise an error. This is how the rest of the checks operate.